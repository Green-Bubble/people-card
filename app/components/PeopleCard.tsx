import { useState } from 'react';
import { People } from '../models/people';
import React from 'react';
import { snakeToCamel } from '../utils';
import Image from 'next/image';
const defaultShow = [
  'first_name',
  'last_name',
  'gender',
  'date_of_birth',
  'avatar',
];

export const PeopleCard = ({ people }: { people: People }) => {
  const [isHover, setIsHover] = useState(false);
  return (
    <>
      <div
        className="card"
        onMouseEnter={() => {
          console.log('mouse entered123123');
          setIsHover(true);
        }}
        onMouseLeave={() => {
          setIsHover(false);
        }}
      >
        <Image
          src={people.avatar}
          alt="Avatar"
          className="dark:invert"
          width={100}
          height={24}
        />
        <h2>{people.first_name + ' ' + people.last_name}</h2>
        <h2>{people.gender}</h2>
        <h2>{people.date_of_birth}</h2>

        {isHover ? (
          <>
            {Object.keys(people)
              .filter(
                key =>
                  defaultShow.indexOf(key) == -1 &&
                  typeof (people as any)[key] === 'string',
              )
              .map(key => (
                <div key={key}>
                  {snakeToCamel(key)} : {(people as any)[key]}
                </div>
              ))}
          </>
        ) : (
          <></>
        )}
      </div>
    </>
  );
};
