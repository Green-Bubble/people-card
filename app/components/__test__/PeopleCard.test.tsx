import renderer from 'react-test-renderer';
import {
  cleanup,
  render,
  screen,
  renderHook,
  act,
  fireEvent,
} from '@testing-library/react';
// import fireEvent from '@testing-library/user-event'
import { PeopleCard } from '../PeopleCard';
import useCounter from '../userCounter';

// jest.mock('../PeopleCard')

const PEOPLE_DATA = {
  id: 712,
  uid: '32bca3d9-8e59-4112-96a1-427aa0becf9d',
  password: 'eEzZUPuc7x',
  first_name: 'Lenard',
  last_name: 'Becker',
  username: 'lenard.becker',
  email: 'lenard.becker@email.com',
  avatar: 'https://robohash.org/veniamquiaest.png?size=300x300\u0026set=set1',
  gender: 'Non-binary',
  phone_number: '+91 269.293.4385 x86321',
  social_insurance_number: '708732342',
  date_of_birth: '1965-07-05',
  employment: {
    title: 'Internal Design Executive',
    key_skill: 'Leadership',
  },
  address: {
    city: 'East Edisonland',
    street_name: 'Schuppe Manor',
    street_address: '150 Fahey Drive',
    zip_code: '60804',
    state: 'Ohio',
    country: 'United States',
    coordinates: { lat: 0.365652409936601, lng: -12.672722971666047 },
  },
  credit_card: { cc_number: '5357-0715-3802-7995' },
  subscription: {
    plan: 'Platinum',
    status: 'Blocked',
    payment_method: 'Cash',
    term: 'Payment in advance',
  },
};

describe('PeopleCard Component', () => {
  beforeAll(() => {
    console.log('Initialize something before all');
  });
  it('changes the class when hovered', () => {
    const component = renderer.create(
      <PeopleCard people={PEOPLE_DATA}></PeopleCard>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
      tree.props.onMouseEnter();
    });

    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
      tree.props.onMouseLeave();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  afterEach(cleanup);

  it('PeopleCard changes the text when mouse hover', () => {
    const res = render(<PeopleCard people={PEOPLE_DATA} />);
    expect(res.queryByText('Non-binary')).toBeTruthy();

    expect(res.queryByText('708732342')).toBeNull();

    fireEvent.mouseEnter(res.container);

    // expect(res.queryByText('708732342')).toBeTruthy();
    // fireEvent.mouseLeave(res.container)

    // expect(res.queryByText('708732342')).toBeNull();
  });

  test('should increment counter', () => {
    const { result } = renderHook(() => useCounter());

    act(() => {
      result.current.increment();
    });

    expect(result.current.count).toBe(1);
  });

  afterAll(() => {
    console.log('Release something before all');
  });
});
