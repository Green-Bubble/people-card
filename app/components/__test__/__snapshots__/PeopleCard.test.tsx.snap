// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCard Component changes the class when hovered 1`] = `
<div
  className="card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Avatar"
    className="dark:invert"
    data-nimg="1"
    decoding="async"
    height={24}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=256&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <h2>
    Lenard Becker
  </h2>
  <h2>
    Non-binary
  </h2>
  <h2>
    1965-07-05
  </h2>
</div>
`;

exports[`PeopleCard Component changes the class when hovered 2`] = `
<div
  className="card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Avatar"
    className="dark:invert"
    data-nimg="1"
    decoding="async"
    height={24}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=256&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <h2>
    Lenard Becker
  </h2>
  <h2>
    Non-binary
  </h2>
  <h2>
    1965-07-05
  </h2>
  <div>
    Uid
     : 
    32bca3d9-8e59-4112-96a1-427aa0becf9d
  </div>
  <div>
    Password
     : 
    eEzZUPuc7x
  </div>
  <div>
    Username
     : 
    lenard.becker
  </div>
  <div>
    Email
     : 
    lenard.becker@email.com
  </div>
  <div>
    PhoneNumber
     : 
    +91 269.293.4385 x86321
  </div>
  <div>
    SocialInsuranceNumber
     : 
    708732342
  </div>
</div>
`;

exports[`PeopleCard Component changes the class when hovered 3`] = `
<div
  className="card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Avatar"
    className="dark:invert"
    data-nimg="1"
    decoding="async"
    height={24}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=256&q=75"
    srcSet="/_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Frobohash.org%2Fveniamquiaest.png%3Fsize%3D300x300%26set%3Dset1&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <h2>
    Lenard Becker
  </h2>
  <h2>
    Non-binary
  </h2>
  <h2>
    1965-07-05
  </h2>
</div>
`;
